pkg/receive/hashring.go

```diff
@@ -261,14 +261,7 @@ func newMultiHashring(algorithm HashringAlgorithm, replicationFactor uint64, cfg
 	return m
 }
 
-// HashringFromConfigWatcher creates multi-tenant hashrings from a
-// hashring configuration file watcher.
-// The configuration file is watched for updates.
-// Hashrings are returned on the updates channel.
-// Which hashring to use for a tenant is determined
-// by the tenants field of the hashring configuration.
-// The updates chan is closed before exiting.
-func HashringFromConfigWatcher(ctx context.Context, algorithm HashringAlgorithm, replicationFactor uint64, updates chan<- Hashring, cw *ConfigWatcher) error {
+func HashringConfigFromConfigWatcher(ctx context.Context, updates chan<- []HashringConfig, cw *ConfigWatcher) error {
```

Comment:
```suggestion
func HashringConfigFromWatcher(ctx context.Context, updates chan<- []HashringConfig, cw *ConfigWatcher) error {
```

Reply:
I'll move this to `config.go`. There're some private functions that we made it public. I move them upward too.